clients/admin-ui/src/features/datamap/datamap.slice.ts

```diff
@@ -132,6 +132,44 @@ const initialState: SettingsState = {
   isGettingStarted: false,
 };
 
+const mergeColumns = (
```

Comment:
this would be a great candidate for a jest test! to make it easier to test, you could say the func only really needs `value` (which also makes the need to cast via `as` go away, I think)

```typescript
const mergeColumns = <T extends {value: string}>(
  columns: T[] | undefined,
  updatedColumns: T[]
) => {
 ...
}

// test, could parametrize with .each()
it("merges properly", () => {
  const columns = [{value: "removed field"}];
  const updatedColumns = [];
  expect(mergeColumns(columns, updatedColumns)).toEqual([])
})

```